semanticdb/metap/src/main/scala/scala/meta/internal/metap/SyntheticPrinter.scala

```diff
@@ -1,28 +1,84 @@
 package scala.meta.internal.metap
 
-import scala.math.Ordering
-import scala.meta.internal.semanticdb._
+import scala.meta.internal.{semanticdb => s}
 
-trait SyntheticPrinter extends BasePrinter with OccurrencePrinter {
-  def pprint(synth: Synthetic): Unit = {
+trait SyntheticPrinter
+    extends BasePrinter
+    with RangePrinter
+    with SymbolInformationPrinter
+    with ConstantPrinter {
+  self =>
+
+  private def shorten(s: String): String = {
+    val newline = s.indexOf('\n')
+    if (newline == -1) {
+      if (s.length > 100) s.takeRight(100) + "..."
+      else s
+    } else s.substring(0, newline) + "..."
+  }
+
+  def pprint(synth: s.Synthetic): Unit = {
     opt(synth.range)(pprint)
-    opt(": ", doc.substring(synth.range))(out.print)
+    opt(": orig(", doc.substring(synth.range), ")")(out.print)
     out.print(" => ")
-    synth.text match {
-      case Some(text) =>
-        out.println(text.text)
-        rep("  ", text.occurrences.sorted, "  ") { occ =>
-          opt(occ.range)(pprint)
-          opt(": ", text.substring(occ.range))(out.print)
-          pprint(occ.role)
-          out.println(occ.symbol)
-        }
-      case _ =>
-        out.println("<?>")
+    pprint(synth.tree)
+  }
+
+  def pprint(tree: s.Tree): Unit = {
+    val infoNotes = new InfoNotes
+    val treePrinter = new TreePrinter(infoNotes)
+    treePrinter.pprint(tree)
+    out.println()
+
+    if (settings.format.isDetailed) {
+      printInfosVisited(infoNotes.visited)
     }
   }
 
-  implicit def synthOrder: Ordering[Synthetic] = {
-    Ordering.by(s => (s.range, s.text.map(_.text)))
+  implicit def newSynthOrder: Ordering[s.Synthetic] = Ordering.by(_.range)
+
+  private class TreePrinter(notes: InfoNotes) extends InfoPrinter(notes) {
+
+    def pprint(tree: s.Tree): Unit = {
+      tree match {
+        case tree: s.ApplyTree =>
+          pprint(tree.fn)
+          out.print("(")
+          rep(tree.args, ",")(pprint)
```

Comment:
The Scalameta tree printer (as well as related printers in other Scala compilers) use `, ` (comma and whitespace) to separate arguments. Let's do that too.